packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.tsx

```diff
@@ -145,10 +145,9 @@ export class ComboBox extends BaseComponent<IComboBoxProps, IComboBoxState> {
   }
 
   public componentWillReceiveProps(newProps: IComboBoxProps) {
-    // In controlled component usage where selectedKey or value is provided,
-    // update the selectedIndex and currentOptions state if the selectedKey or options have changed
-    if ((newProps.selectedKey || newProps.value) &&
-      (newProps.selectedKey !== this.props.selectedKey || newProps.options !== this.props.options)) {
+    // Update the selectedIndex and currentOptions state if the selectedKey or options have changed
+    if (((newProps.selectedKey || newProps.value) && newProps.selectedKey !== this.props.selectedKey) ||
+      newProps.options !== this.props.options) {
```

Comment:
is the second half of this if statement sufficient (i.e. just checking if the props are different vs checking that they also are truthy)?

Reply:
Yes, I think it's sufficient, it is better to update the currentOptions when the objects are different as opposed to trying to do something smarter here and potentially missing a case.